DAL/EmployeeDAL.cs

```diff
@@ -0,0 +1,218 @@
+using EmployeeManagementSystem.Utils;
+using EmployeeManagementSystem.Models;
+namespace EmployeeManagementSystem.DAL;
+public class EmployeeDAL : IEmployeeDAL
+{
+    public string filePath = "";
+    public readonly JSONUtils jsonUtils;
+    public readonly ILogger _logger;
+
+    public EmployeeDAL(ILogger logger, JSONUtils jsonUtils, string filePath)
+    {
+        this.jsonUtils = jsonUtils;
+        _logger = logger;
+        this.filePath = filePath;
+    }
+
+    public bool Insert(Employee employee)
+    {
+        List<Employee> existingEmployees = jsonUtils.ReadJSON<Employee>(filePath);
+        existingEmployees.Add(employee);
+        jsonUtils.WriteJSON(existingEmployees, filePath);
+        return true;
+    }
+
+    public (List<EmployeeDetails>? employees, bool status) RetrieveAllEmployees()
+    {
+        List<Employee> existingEmployees = jsonUtils.ReadJSON<Employee>(filePath);
+        if (existingEmployees == null || existingEmployees.Count == 0)
+        {
+            return (null, false);
+        }
+        List<EmployeeDetails> employeeDetailsList = new List<EmployeeDetails>();
+
+        foreach (var employee in existingEmployees)
+        {
+            EmployeeDetails employeeDetails = ConvertToEmployeeDetails(employee);
+            employeeDetailsList.Add(employeeDetails);
+        }
+        return (employeeDetailsList, true);
+    }
+
+    public bool Update(Employee updatedEmployee, string empNo)
+    {
+        List<Employee> existingEmployees = jsonUtils.ReadJSON<Employee>(filePath);
+        Employee employeeToUpdate = existingEmployees.FirstOrDefault(emp => emp.EmpNo == empNo);
+
+        if (employeeToUpdate == null)
+        {
+            return false;
+        }
+
+        // Update employee fields
+        employeeToUpdate.FirstName = updatedEmployee.FirstName == "--d" ? null :
```

Comment:
this logic is not good. instead expect a new list<string> that contains properties to be updated